### PR TITLE
Switch to CircleCI 2

### PR DESCRIPTION
**Pull request by SerialVelocity**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      _JAVA_OPTIONS: '-XX:ActiveProcessorCount=8 -Xmx2048m'
+    steps:
+      - checkout
+      - restore_cache: { key: 'gradle-wrapper-{{ checksum "gradle/wrapper/gradle-wrapper.properties" }}' }
+      - restore_cache: { key: 'gradle-cache-{{ checksum "versions.props" }}-{{ checksum "build.gradle" }}' }
+      - setup_remote_docker:
+          docker_layer_caching: true
+      - run:
+          name: Make remote docker appear as local docker
+          command: sudo ssh -f -o "ExitOnForwardFailure yes" -F $HOME/.ssh/config -4N -L /var/run/docker.sock:/var/run/docker.sock $(for port in 2376 $(seq 32768 32999); do echo "-L ${port}:127.0.0.1:${port}"; done) remote-docker
```
Comment (iamdanfox) [inline .circleci/config.yml]: Personally, I'd recommend using the `machine` executor instead of wiring up lots of funky SSH stuff.

Comment (SerialVelocity) [reply .circleci/config.yml]: Any idea how to get Java 11 working on the `machine` executor?

Comment (iamdanfox) [reply .circleci/config.yml]: You can just download openjdk11 and untar it.

Comment (SerialVelocity) [reply .circleci/config.yml]: I'm going to merge as-is for now to get Java 11 support in and probably update the java-library-oss template in the future at some point instead.

Comment (SerialVelocity) [reply .circleci/config.yml]: nvm, I found a simple example to copy so I've switched it to machine executors now.

**Pull request merged**